e3db-crypto-plain/src/main/java/com/tozny/e3db/PlainCrypto.java

```diff
@@ -70,69 +70,69 @@ public CipherSuite suite() {
   }
 
   @Override
-  public CipherWithNonce encryptSecretBox(byte[] message, byte[] key) {
+  public CipherWithNonce encryptSecretBox(byte[] message, byte[] key) throws E3DBEncryptionException {
     checkNotNull(message, "message");
     checkNotEmpty(key, "key");
     byte[] nonce = lazySodium.randomBytesBuf(SecretBox.NONCEBYTES);
     byte[] cipher = new byte[SecretBox.MACBYTES + message.length];
     if(! lazySodium.cryptoSecretBoxEasy(cipher, message, message.length, nonce, key))
-      throw new RuntimeException("Could not encrypt message.");
+      throw new E3DBEncryptionException("Could not encrypt message.");
 
     return new CipherWithNonce(cipher, nonce);
   }
 
   @Override
-  public byte[] decryptSecretBox(CipherWithNonce message, byte[] key) {
+  public byte[] decryptSecretBox(CipherWithNonce message, byte[] key) throws E3DBDecryptionException {
     checkNotNull(message, "message");
     checkNotEmpty(key, "key");
     byte[] messageBytes = new byte[message.getCipher().length - SecretBox.MACBYTES];
     if(! lazySodium.cryptoSecretBoxOpenEasy(messageBytes, message.getCipher(), message.getCipher().length, message.getNonce(), key))
-      throw new RuntimeException("Could not decrypt message.");
+      throw new E3DBDecryptionException("Could not decrypt message.");
```

Comment:
Not blocking or critical, but might save our future selves some headaches if we  extract string error messages like "Could not decrypt" that get used multiple places into a constant.

Reply:
That's a good thought. I can do that

Reply:
Yes -- this is very good practice. Also - when we're a super popular international library that needs to translate strings to different languages, having that abstraction is quite helpful ;)